### PR TITLE
feat: stopwatch for task duration, single task page

### PR DESCRIPTION
**Pull request by HarshnaNHaswani**
image
image
image
image
image

### CODE REVIEW

File: src/components/StopWatch.jsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import PlayDark from "../assets/playDark.png";
+import RestartDark from "../assets/restartDark.png";
+import StopDark from "../assets/stopDark.png";
+import { useTasks } from "../context/task-context";
+import { useTheme } from "../context/theme-context";
+
+export const StopWatch = ({ duration, taskId, disabled }) => {
+  const { completeTask } = useTasks();
+  const { theme } = useTheme();
+  const {dark} = theme;
```
Comment (HeyNitin) [inline src/components/StopWatch.jsx]: you can destructure it in one line, {theme: {dark}} = useTheme()

File: src/components/StopWatch.jsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  const timerRef = useRef({ timerId: null, currentDuration: null });
+  const totalSeconds = timerRef.currentDuration
+    ? timerRef.currentDuration
+    : duration.hours * 60 * 60 + duration.minutes * 60 + duration.seconds;
+  const [progress, setProgress] = useState(totalSeconds);
+  const [isPaused, setIsPaused] = useState(true);
+
+  const start = () => {
+    timerRef.current.timerId = setInterval(() => {
+      setIsPaused(false);
```
Comment (HeyNitin) [inline src/components/StopWatch.jsx:25]: does setIsPaused need to be updated every second ? or only once. if once, it should be outside interval, even though it doesn't trigger re render, it is not good coding practice.

File: src/pages/SingleTask/SingleTask.jsx [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+import { StopWatch } from "../../components/StopWatch";
+import { useTasks } from "../../context/task-context";
+import { findTask } from "../../utils/findTask";
+import NotPinned from "../../assets/notPinned.png";
+import NotPinnedDark from "../../assets/notPinnedDark.png";
+import Archive from "../../assets/archive.png";
+import ArchiveDark from "../../assets/archiveDark.png";
+import Unarchive from "../../assets/unarchive.png";
+import UnarchiveDark from "../../assets/unarchiveDark.png";
+import Restore from "../../assets/restore.png";
```
Comment (HeyNitin) [inline src/pages/SingleTask/SingleTask.jsx]: Just a suggestion: You can have a look at absolute imports, they make life easier, especially in big projects

**Review by HeyNitin — APPROVED** [review]
Left some suggestions, rest looks good, really liked the UI of the app, great work there

**Review by mansipadgelwar — APPROVED** [review]
The code is well structured. Looks good.

**Review by simran61 — APPROVED** [review]
Looks good. Clean folder structure

File: src/components/StopWatch.jsx (hunk 1 above)
Comment (HarshnaNHaswani) [reply src/components/StopWatch.jsx]: I didn't know that, thanks for letting me know

File: src/components/StopWatch.jsx (hunk 2 above)
Comment (HarshnaNHaswani) [reply src/components/StopWatch.jsx:25]: only once, I must have missed it thanks for pointing it out

File: src/pages/SingleTask/SingleTask.jsx (hunk 3 above)
Comment (HarshnaNHaswani) [reply src/pages/SingleTask/SingleTask.jsx]: Read about it, it's awesome, thanks!

**Comment by HarshnaNHaswani** [comment]
thanks for pr @HeyNitin  @mansipadgelwar @simran61.  I'll make the necessary changes

**Pull request merged**